openfoodfacts/images.py

```diff
@@ -254,3 +254,99 @@ def download_image(
         session=session,
         return_struct=return_struct,
     )
+
+
+def convert_to_legacy_schema(images: JSONType) -> JSONType:
+    """Convert the images dictionary to the legacy schema.
+
+    We've improved the schema of the `images` field, but the new
+    schema is not compatible with the legacy schema. This function
+    converts the new schema to the legacy schema.
+
+    It can be used while migrating the existing Python codebase to the
+    new schema.
+
+    The new `images` schema is the following:
+
+    - the `images` field contains the uploaded images under the `uploaded`
+        key and the selected images under the `selected` key
+    - `uploaded` contains the images that are uploaded, and maps the
+        image ID to the detail about the image:
+        - `uploaded_t`: the upload timestamp
+        - `uploader`: the username of the uploader
+        - `sizes`: dictionary mapping image size (`100`, `200`, `400`, `full`)
+            to the information about each resized image:
+            - `h`: the height of the image
+            - `w`: the width of the image
+            - `url`: the URL of the image
+    - `selected` contains the images that are selected, and maps the
+        image key (`nutrition`, `ingredients`, `packaging`, or `front`) to
+        a dictionary mapping the language to the selected image details.
+        The selected image details are the following fields:
+        - `imgid`: the image ID
+        - `rev`: the revision ID
+        - `sizes`: dictionary mapping image size (`100`, `200`, `400`, `full`)
+            to the information about each resized image:
+            - `h`: the height of the image
+            - `w`: the width of the image
+            - `url`: the URL of the image
+        - `generation`: information about how to generate the selected image
+            from the uploaded image:
+            - `geometry`
```

Comment:
I removed the geometry field in the new schema (it was used only as a parameter to ImageMagick). It could be derived from the coordinates if you want to generate it back, but I'm guessing nobody was using it.